src/main/cpp/src/json_parser.cuh

```diff
@@ -1386,20 +1386,22 @@ class json_parser {
       return true;
     }
 
+    json_token t;
     int open = 1;
-    while (true) {
-      json_token t = next_token();
+    do {
+      t = next_token();
       if (t == json_token::START_OBJECT || t == json_token::START_ARRAY) {
         ++open;
       } else if (t == json_token::END_OBJECT || t == json_token::END_ARRAY) {
         if (--open == 0) { return true; }
       } else if (t == json_token::ERROR) {
         return false;
       }
-    }
+    } while (t != json_token::SUCCESS);
```

Comment:
Question: It looks like a logic change to check if `t != json_token::SUCCESS`. Was there a bug?

Reply:
It should not be a bug. Digging into the code, for normal situations it will return `true` and for situations like `[[[[[[[[.......` I see that this will ultimately result in the `ERROR` token. 

However, a better logic IMO is to have a return at the end of the function so I changed into this. Essentially, the behavior of this function does not change at all.